src/bst/core.clj

```diff
@@ -109,55 +113,57 @@
 
     :else tree))
 
-(defn insert-node
+(defn insert-and-balance
```

Comment:
Think about whether the word `balance` is needed here. The fact that the tree is balanced is an implementation detail.

Reply:
This function was named insert-node initially. But when I was asked to write separate functions for insert and balance, I renamed this insert-and-balance and named the function which only inserted a node to the tree as insert-node.
Since I don't have that new functions anymore, I will change it back to insert-node.